python/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+use pyo3::{
+    exceptions::{PyException, PyIOError, PyValueError},
+    prelude::*,
+    types::PyDict,
+};
+use std::path::PathBuf;
+
+/// A CQL2 expression.
+#[pyclass]
+struct Expr(::cql2::Expr);
+
+/// A SQL query
+#[pyclass]
+struct SqlQuery {
+    #[pyo3(get)]
+    query: String,
+
+    #[pyo3(get)]
+    params: Vec<String>,
+}
+
+#[pymethods]
+impl Expr {
+    /// Parses a CQL2 expression from a file path.
+    #[staticmethod]
+    fn from_path(path: PathBuf) -> PyResult<Expr> {
+        ::cql2::parse_file(path).map(Expr).map_err(to_py_error)
+    }
+
+    /// Parses a CQL2 expression.
+    #[new]
+    fn new(cql2: Bound<'_, PyAny>) -> PyResult<Self> {
+        if let Ok(s) = cql2.extract::<&str>() {
+            s.parse().map(Expr).map_err(to_py_error)
+        } else {
+            let expr: ::cql2::Expr = pythonize::depythonize(&cql2)?;
+            Ok(Expr(expr))
+        }
+    }
+
+    /// Converts this expression to a cql2-json dictionary.
+    fn to_json<'py>(&self, py: Python<'py>) -> PyResult<Bound<'py, PyDict>> {
```

Comment:
You don't really gain at all from having `PyDict` in the return type here, when `pythonize` returns a `Result<Bound<'py, PyAny>>`, so I personally would return a `Result<Bound<'py, PyAny>>` and remove the `any.extract()` line